src/services/refactors/moveToFile.ts

```diff
@@ -946,7 +946,8 @@ export function getUsageInfo(oldFile: SourceFile, toMove: readonly Statement[],
     }
 }
 
-function isGlobalType(checker: TypeChecker, symbol: Symbol) {
+function isGlobalType(checker: TypeChecker, location: Node, symbol: Symbol) {
+    if (checker.resolveName(symbol.name, location, SymbolFlags.All, /*excludeGlobals*/ true)) return false;
     return !!checker.resolveName(symbol.name, /*location*/ undefined, SymbolFlags.Type, /*excludeGlobals*/ false);

```

Comment:
I'm having trouble wrapping my head around this function, I think because it takes a symbol but only uses it for its name. Do the function name and parameters need to be tweaked a bit to better reflect what this is trying to accomplish?

Reply:
Is there another way to check if a symbol refers to a global, and can be checked reliably after symbol merging/through aliasing? I guess this function doesn't need a Symbol to be passed in as it currently is, but I couldn't think of a better way to do this check for globals.

The old code was checking if there was _any_ global symbol with that name that was a type without seeing first if there was a local symbol with that name. The function name seems alright to me since it's checking if, at this node, the symbol refers to a global type.

Reply:
> But the line added here seems to allow for a false negative of the simple question “does this name refer to a global type.” Just because a name resolves to a local value doesn’t mean it doesn’t also resolve to a global type at the same location:

The correct check to have here is if the _symbol_ refers to a global type or not, and the previous check did not do that (and seems like the previous check wanted to check the symbol, given that they took a symbol instead of just the name).

> even if it has the desired behavior for Move To File (does it?).

It makes sense to me that if we try to move a symbol that references a local definition, but also happens to have an unrelated global definition, that we keep the reference in the new location as the same symbol that was in the original file. Otherwise, the code in the new file can be broken, like in the bug report above.  I spoke to @navya9singh about this, and we concluded that we should generate the import to keep the original reference. Also, other codefixes keep local references across files, even if the symbol resolves to a global (added test to show this).

Reply:
@andrewbranch am I correct that you're mostly not a fan of the function doing the following?

1. taking a `Symbol` instead of a `string` (since all it needs is the name of the symbol, and
2. being named `isGlobalType`, since a type might not actually be declared in the global scope (so e.g. `cannotResolveTypeAt` would be a "better" name)

Reply:
Yes - I didn't recontextualize myself with the broader problem enough to tell whether the implementation/behavior is wrong; I can just tell from reading the function in isolation that something is a little fishy (specifically, your points above).

Reply:
@DanielRosenwasser @andrewbranch How do you feel about the new behavior